ibm/service/power/resource_ibm_pi_network_address_group.go

```diff
@@ -0,0 +1,247 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceIBMPINetworkAddressGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPINetworkAddressGroupCreate,
+		ReadContext:   resourceIBMPINetworkAddressGroupRead,
+		UpdateContext: resourceIBMPINetworkAddressGroupUpdate,
+		DeleteContext: resourceIBMPINetworkAddressGroupDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_Name: {
+				Description:  "The name of the Network Address Group.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_UserTags: {
+				Description: "The user tags associated with this resource.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Set:         schema.HashString,
+				Type:        schema.TypeSet,
```

Comment:
The IBM Terraform Provideris used in Schematics to provision templates or deployable architecture. When we provision infrastructure using Schematics it add some additional tags like workspaceid..so on.
In order to suppress the diff we need to add this custom diff

```CustomizeDiff: customdiff.Sequence(
			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
				return flex.ResourceTagsCustomizeDiff(diff)
			},
		),
		```
		we need to add this custom diff for all power resources where we have tags introdued

Reply:
@hkantare Could elaborate more on this where do we add them or give an example of this?

We are using `flex.UpdateGlobalTagsUsingCRN` which uses `Set` from what a user is provided through power, then only difference would be if the schematicsTags is the same as the one provided. At that point we can not control that part. In `flex.UpdateGlobalTagsUsingCRN` tags are appended to the list a user provided.
```
schematicTags := os.Getenv("IC_ENV_TAGS")
		var envTags []string
		if schematicTags != "" {
			envTags = strings.Split(schematicTags, ",")
			add = append(add, envTags...)
		}
```

Reply:
NVM -- I found the example.